src/components/policy/policy_external/src/update_status_manager.cc

```diff
@@ -121,9 +121,11 @@ void UpdateStatusManager::OnResetRetrySequence() {
 void UpdateStatusManager::OnNewApplicationAdded(const DeviceConsent consent) {
   LOG4CXX_AUTO_TRACE(logger_);
   if (kDeviceAllowed != consent) {
+    LOG4CXX_DEBUG(logger_, "Application registered from non-consented device");
     app_registered_from_non_consented_device_ = true;
```

Comment:
Why app_registered_from_non_consented_device_ == true by default (in constructor)? It would be incorrect for consented device without applications, in this case SDL sends OnStatusUpdate(UPDATE_NEEDED) in OnDeviceConsented() without triggers

Reply:
@APipko It is impossible case for real usage.

Reply:
@AByzhynar this case is not realistic, but not impossible. Setting `app_registered_from_non_consented_device_ = true` by default is illogical and confusing. Also sending redundant OnStatusUpdate(UPDATE_NEEDED) misleads during analyzing ATF test scripts logs.

Reply:
@APipko Detailed flow, please

Reply:
@AByzhynar For example https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/RC/CLIMATE_RADIO/Capabilities/001_All_modules_all_params.lua#L34
Device is consented without registered applications

Reply:
@APipko Check your link again!!! CAREFULLY!!!! There is 1 application REGISTERED and ACTIVATED!!!!

Reply:
@AByzhynar As far as I understand, registration and activation of the application are performed in the next lines. In line#34 only SDL started, HMI initialized, mobile connected and consented. And right after that OnStatusUpdate(UPDATE_NEEDED) is sent.
``` 
HMI->SDL : "method":"SDL.OnAllowSDLFunctionality"
MOB->SDL : controlMsg: StartService
SDL->HMI : "method":"BasicCommunication.UpdateDeviceList"
HMI->SDL : "result":{"method":"BasicCommunication.UpdateDeviceList","code":0}
SDL->HMI : "method":"SDL.OnStatusUpdate","params":{"status":"UPDATE_NEEDED"}
SDL->MOB : controlMsg: StartServiceACK 
MOB->SDL : rpcFunction: RegisterAppInterface 
```